src/Agent.Listener/MessageListener.cs

```diff
@@ -179,5 +235,10 @@ public sealed class MessageListener : AgentService, IMessageListener
                 return message;
             }
         }
+
+        private bool IsFatalException(Exception ex)
```

Comment:
I want to catch both fatal and non-fatal exceptions and have only partial difference in the way I handle them.